### ISSUE TITLE
Source: Piwik plugin for Lagotto 4.3

### ISSUE THREAD

**Issue by sachbearbeiter** [lagotto/lagotto#653]
Hello,
we have a client that uses Lagotto in the version 4. and wants to feed his system also with data from Piwik.
Has someone maybe programmed a connector to a local Piwik API (or db)?
If someone who is familiar with Lagotto and Ruby could do a paid programming for us, this would be also be wonderful ...

You can contact me via post@diesachbearbeiter.de

If this issue is out of place here please give me a short note and delete this ...

Thanks and regards
Immo

**Comment by afandian** [comment]
I am interested to hear exactly what format the data comes in. Is it page impressions per DOI, or per article landing page URL?

**Comment by sachbearbeiter** [comment]
Thanks for the reply ...

I don't understand exactly your question, but i think the Piwik is collecting the visits (page impressions etc.) "per article landing page URL". I have no deep understanding of Piwik nor Lagotto: Do you mean Piwik could be configured to collect the visits by DOI (some kind of campaign ...)?

Regards
Immo

**Comment by afandian** [comment]
Sorry if I wasn't clear. Lagotto collects data per Work, and each work has an identifier, usually the DOI. It can also have a URL, for example, the landing page where the article is displayed.

Example of a landing page URL: http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0174371
Example of the DOI for the same work: https://doi.org/10.1371/journal.pone.0174371

Lagotto will usually use the DOI to identify the Work.

My question is, does Piwik report per landing page URL (where people visit, and where I, would imagine, the Piwik tracker reports the visit) or does it internally convert this into the DOI before sending to Lagotto?

Is your client interested in making their Piwik data open to the public, or is it confidential?

**Comment by sachbearbeiter** [comment]
> My question is, does Piwik report per landing page URL (where people visit, and where I, would imagine, the Piwik tracker reports the visit) or does it internally convert this into the DOI before sending to Lagotto?

> Is your client interested in making their Piwik data open to the public, or is it confidential?

I forwarded your questions to the client ...

**Comment by sachbearbeiter** [comment]
The answer of the client:

> Piwik reports it per landing page URL and does not convert it to DOI. While of course we don’t want to make the entirety of our Piwik data public (e.g. referrers, location, browsers), we will certainly make the number of visits and visits per date public for each article / tracked page.

Does this answer your question sufficiently?

**Comment by afandian** [comment]
Yes. I can't help directly with Lagotto, but this may be a good fit for Crossref Event Data, which integrates with Lagotto. Contact me at jwass@crossref.org if you want to discuss.

**Comment by sachbearbeiter** [comment]
Thanks for the feedback ...
The client will contact you directly with further questions ;)

If someone else can programme this Plugin for us - feel free to contact me ...
